pkg/proxy/ipvs/proxier.go

```diff
@@ -808,6 +808,42 @@ func (proxier *Proxier) syncProxyRules() {
 	// activeBindAddrs represents ip address successfully bind to DefaultDummyDevice in this round of sync
 	activeBindAddrs := map[string]bool{}
 
+	hasNodePort := false
+	for _, svc := range proxier.serviceMap {
+		svcInfo, ok := svc.(*serviceInfo)
+		if ok && svcInfo.NodePort() != 0 {
+			hasNodePort = true
+			break
+		}
+	}
+
+	// Both nodeAddresses and nodeIPs can be reused for all nodePort services
+	// and only need to be computed if we have at least one nodePort service.
+	var (
+		// List of node addresses to listen on if a nodePort is set.
+		nodeAddresses sets.String
+		// List of node IP addresses to be used as IPVS services if nodePort is
+		// set.
+		nodeIPs []net.IP
+	)
+
+	if hasNodePort {
+		nodeAddresses, err = utilproxy.GetNodeAddresses(proxier.nodePortAddresses, proxier.networkInterfacer)
+		if err != nil {
+			klog.Errorf("Failed to get node ip address matching nodeport cidr: %v", err)
+		}
+		for address := range nodeAddresses {
+			if utilproxy.IsZeroCIDR(address) {
+				nodeIPs, err = proxier.ipGetter.NodeIPs()
+				if err != nil {
+					klog.Errorf("Failed to list all node IPs from host, err: %v", err)
+				}
+				break
```

Comment:
Is this `break` intentional?

Reply:
Yes, that's intentional. The reasoning was that if `IsZeroCIDR` is true to any address then `nodeIPs` will be set to all host IPs (the return of `ipGetter.NodeIPs()`) and we don't need to append extra IP addresses to `nodeIPs` and it's safe to break here.